src/components/expanded-state/AvailableNetworksv2.tsx

```diff
@@ -62,10 +72,65 @@ const AvailableNetworksv2 = ({
       // we need to convert the mainnet asset to the selected network's
       newAsset.mainnet_address = networks?.[ethereumUtils.getChainIdFromNetwork(Network.mainnet)]?.address ?? asset.address;
       newAsset.address = networks?.[ethereumUtils.getChainIdFromNetwork(chosenNetwork)].address;
+      newAsset.network = chosenNetwork;
+
+      goBack();
+
+      if (swapsV2Enabled || swaps_v2) {
+        const chainId = ethereumUtils.getChainIdFromNetwork(newAsset.network);
+        const uniqueId = `${newAsset.address}_${chainId}`;
+        const userAsset = userAssetsStore.getState().userAssets.get(uniqueId);
+
+        console.log(chainId, uniqueId);
+
+        const parsedAsset = parseSearchAsset({
+          assetWithPrice: {
+            ...newAsset,
+            uniqueId,
+            address: newAsset.address as AddressOrEth,
+            type: newAsset.type as AssetType,
+            chainId,
+            chainName: chainNameFromChainId(chainId),
+            isNativeAsset: false,
+            native: {},
+          },
+          searchAsset: {
+            ...newAsset,
+            uniqueId,
+            chainId,
+            chainName: chainNameFromChainId(chainId),
+            address: newAsset.address as AddressOrEth,
+            highLiquidity: newAsset.highLiquidity ?? false,
+            isRainbowCurated: newAsset.isRainbowCurated ?? false,
+            isVerified: newAsset.isVerified ?? false,
+            mainnetAddress: (newAsset.mainnet_address ?? '') as AddressOrEth,
+            networks: newAsset.networks ?? [],
+            type: newAsset.type as AssetType,
+          },
+          userAsset,
+        });
+
+        const largestBalanceSameChainUserAsset = userAssetsStore
+          .getState()
+          .getUserAssets()
+          .find(userAsset => userAsset.chainId === chainId && userAsset.address !== newAsset.address);
+        if (largestBalanceSameChainUserAsset) {
+          swapsStore.setState({ inputAsset: largestBalanceSameChainUserAsset });
+        } else {
+          swapsStore.setState({ inputAsset: null });
+        }
```

Comment:
@christianbaroni I can remove this, but I noticed for cases where we don't have a largest balance on the same chain we still default to the largest overall balance. Is that intended? My thinking is it'd make more sense for users to explicitly select an asset so that we aren't making an initial quote fetch that they (probably) don't want.